pkg/service/schema_test.go

```diff
@@ -150,69 +254,84 @@ func TestValidateIntProperty(t *testing.T) {
 	const fieldName = "xyz"
 
 	ips := IntPropertySchema{}
+	// This should fail validation because the value is of the wrong type
 	err := ips.validate(fieldName, "foobar")
 	assert.NotNil(t, err)
 	validationError, ok := err.(*ValidationError)
 	assert.True(t, ok)
 	assert.Equal(t, fieldName, validationError.Field)
+	// This should fail validation because the value is of the wrong type
 	err = ips.validate(fieldName, 3.14)
 	assert.NotNil(t, err)
 	validationError, ok = err.(*ValidationError)
 	assert.True(t, ok)
 	assert.Equal(t, fieldName, validationError.Field)
+	// This should pass validation-- nil == no value == valid
 	err = ips.validate(fieldName, nil)
 	assert.Nil(t, err)
 
 	ips = IntPropertySchema{
 		MinValue: ptr.ToInt64(3),
 	}
-	err = ips.validate(fieldName, 2.0)
+	// This should fail validation because the value is too small
+	err = ips.validate(fieldName, 2)
```

Comment:
You'll see a lot of this. Since validations got more tolerant of what's in the maps vis-a-vis types, we don't have to use floats in the tests where we would have preferred to use ints.